libvast/src/system/accountant.cpp

```diff
@@ -216,7 +219,26 @@ struct accountant_state_impl {
                           const metrics_metadata& meta2, time ts,
                           detail::uds_datagram_sender& dest) {
     auto buf = to_json_line(actor_id, ts, key, x, meta1, meta2);
-    dest.send(std::span<char>{reinterpret_cast<char*>(buf.data()), buf.size()});
+    // Only poll the socket for write readiness if it did not drop the previous
+    // line. Not doing this could increase the average message processing time
+    // of the accountant by timeout value, and just as with any actor, receiving
+    // messages at a higher frequency that they can be processed will eventually
+    // consume all available memory.
+    // The initial timeout of 1 second is somewhat arbitrary. Long enough so
+    // any re-initialization of the listening side would not incur data loss
+    // without being excessive.
+    auto timeout_usec = uds_datagram_sink_dropping ? 0 : 1'000'000;
+    auto delivered = dest.send(
+      std::span<char>{reinterpret_cast<char*>(buf.data()), buf.size()},
+      timeout_usec);
+    if (!delivered) {
+      VAST_WARN("{} failed to write metrics to UDS sink: {}", *self,
+                delivered.error());
+      VAST_WARN("{} disables the UDS metrics sink", *self);
```

Comment:
```suggestion
      VAST_WARN("{} resets the UDS metrics sink", *self);
```

Reply:
Isn't "disable" clearer here? If it says "reset" I would ask to what state?

Reply:
I am thinking of "reset" as something that makes starts again from the beginning. "Disabling" to me sounds like it's no longer available.

Reply:
It does turn off. If we get a hard error there is no point in trying again.

Reply:
Okay, then please disregard my suggestion.